### DISCUSSION TITLE
How to get hostname in routing middleware

### DISCUSSION THREAD

**Discussion by lxccc812** [nuxt/nuxt#29654] [Questions]
I want to get the hostname of the current domain name in the routing middleware.
For example, example.com/index I want to get example.com as the hostname. But after passing through the nginx proxy, I can only get 0.0.0.0:3000.

The following is some information I have printed in the routing middleware, but I did not find the hostname I need.
```ts
export default defineNuxtRouteMiddleware((to) => {
    const headers = useRequestHeaders()
    const country = headers['cf-ipcountry']
    console.log('headers', headers)
    console.log('to', to)
})

```

**Comment by xjccc** [comment]
Something like this?

https://github.com/zernonia/keypress/blob/main/server/middleware/subdomain.ts

**Reply by lxccc812** [reply]
> Something like this? 像这样的东西吗？
> 
> [zernonia/keypress@`main`/server/middleware/subdomain.ts](https://github.com/zernonia/keypress/blob/main/server/middleware/subdomain.ts)

I have tried this, but it doesn't solve my problem. I'm not sure if it's a problem with nginx. I can only get `hostname 0.0.0.0:3001`. This is my nginx configuration.
```conf
server {
	listen 80;
	server_name  example.com;
	charset utf-8;

	location / {
		proxy_pass http://0.0.0.0:3001;
	}
}
```

**Reply by xjccc** [reply]
Is this some nginx config?

```
 proxy_set_header Host $host;
```

**Reply by lxccc812** [reply]
Thanks for your answer, but it still doesn't work, in addition I also used swr for caching and set up varies according to the answer of https://github.com/nuxt/nuxt/issues/26102#issuecomment-1980625833.
